src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/ProjectFileClassifier.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+
+namespace Microsoft.VisualStudio.ProjectSystem
+{
+    internal sealed class ProjectFileClassifier
+    {
+        private readonly string _windows;
+        private readonly string _programFiles86;
+        private readonly string? _programFiles64;
+        private readonly string? _vsInstallationDirectory;
+
+        private string[] _nuGetPackageFolders = Array.Empty<string>();
+        private string? _nuGetPackageFoldersString;
+
+        /// <summary>
+        /// Gets and sets the <c>MSBuildProjectExtensionsPath</c> property value for this project.
+        /// Project files under this folder are considered non-modifiable.
+        /// </summary>
+        /// <remarks>
+        /// Example value: <c>"C:\repos\MySolution\MyProject\obj\"</c>
+        /// </remarks>
+        public string? ProjectExtensionsPath { get; set; }
+
+        /// <summary>
+        /// Gets and sets the paths found in the <c>NuGetPackageFolders</c> property value for this project.
+        /// Project files under any of these folders are considered non-modifiable.
+        /// </summary>
+        /// <remarks>
+        /// Example value: <c>"C:\Users\myusername\.nuget\;D:\LocalNuGetCache\"</c>
+        /// </remarks>
+        public string NuGetPackageFolders
+        {
+            set
+            {
+                if (!string.Equals(_nuGetPackageFoldersString, value, StringComparisons.Paths))
+                {
+                    _nuGetPackageFoldersString = value;
+                    _nuGetPackageFolders = value.Split(Delimiter.Semicolon, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
>value.Split [](http://example.com/codeflow?start=43&length=11)

Passing null in a 2nd call to the setter can throw here, right? Is that the desired behavior?

Reply:
The parameter has a non-nullable type, but validating the argument makes sense. I've added a `Requires.NotNull`.